diff --git a/swg-mods-common b/swg-mods-common
index ed92d55..8586398 100755
--- a/swg-mods-common
+++ b/swg-mods-common
@@ -127,7 +127,7 @@ fi
 
 # use /persist/tee as secure storage path
 # https://github.com/vchong/optee_client/commit/68e907e6c6a05c2e5c8603079dfae5927fce4430
-curl_am_optee optee_client 68e907e6c6a05c2e5c8603079dfae5927fce4430 external/optee_client vchong
+#curl_am_optee optee_client 68e907e6c6a05c2e5c8603079dfae5927fce4430 external/optee_client vchong
 
 if [ X"$TEE_USE_RPMB" = X"true" ]; then
 	# tee-supplicant: disable RPMB emu
@@ -155,7 +155,7 @@ fi
 cd optee/optee_os
 #git fetch https://github.com/vchong/optee_os pull/1/head && git merge --no-edit FETCH_HEAD
 # cherry pick instead of merge so that patches stay on top
-git fetch https://github.com/vchong/optee_os kmgk_rebase_mbedtls && git cherry-pick da2619a4a694f436e0b178aa06ac5f830166e682^..25319ff90d708627eb41a7cad20c650a10e30e10
+git fetch https://github.com/vchong/optee_os kmgk_rebase_mbedtls_380_dnd && git cherry-pick f29b0cf52a53b40141a1485f28aa41368af4171f^..ce6bc815113dc4db0fd25763370bb67be101a9ac
 cd -
 
 #######################################################
