diff --git a/swg-mods-common b/swg-mods-common
index 7d0319f3..cf228512 100755
--- a/swg-mods-common
+++ b/swg-mods-common
@@ -155,7 +155,7 @@ fi
 cd optee/optee_os
 #git fetch https://github.com/vchong/optee_os pull/1/head && git merge --no-edit FETCH_HEAD
 # cherry pick instead of merge so that patches stay on top
-git fetch https://github.com/vchong/optee_os kmgk_rebase_mbedtls && git cherry-pick a4097612cde65f24cc9ba22048dd5708dfb3ffbc^..d054b7fad46b14e669e6aa3226c85bb89a92e24c
+git fetch https://github.com/vchong/optee_os kmgk_rebase_mbedtls_380_dnd && git cherry-pick f29b0cf52a53b40141a1485f28aa41368af4171f^..ce6bc815113dc4db0fd25763370bb67be101a9ac
 cd -
 
 #######################################################
